### PR TITLE
Exposes all "strip markers" arguments

### LINKED ISSUE #16: Stripping ICC Color Profiles leads to some browsers rendering image differently
I noticed an issue when using **jpegoptim** that sometimes my images didn't always appear to be "losslessly" compressed depending on the browser I was using.

After some fiddling around with the binary and running it with my own command-line flags, I determined that stripping the **ICC Color Profile Markers** from the JPG didn't guarantee consistent rendering across browsers. [Doing a quick google search](http://lmgtfy.com/?q=icc+color+profile+browser+render) shows other people reporting this problem (e.g., [here](https://fstoppers.com/education/how-your-web-browser-affects-way-colors-are-rendered-77241), [here](http://cameratico.com/guides/web-browser-color-management-guide/), and [here](https://petapixel.com/2012/06/25/is-your-browser-color-managed/)).

See example:  
icc_stripped_example

However, currently this module doesn't let you control what image markers to strip and instead always passes in the `--strip-all` flag.

The the markers could be individually set, with the default of stripping all, this issue could be resolved.

### PR DESCRIPTION
**Pull request by romellem**
Closes #16 

Bumps minor version since we add new configuration.

### CODE REVIEW

**Comment by romellem** [comment]
Looks like tests are failing? Not sure what to fix there since it also looks like there were failing when they were re-written in #12.

File: readme.md [hunk 1]
```diff
@@ -36,24 +36,17 @@ Returns a promise for a buffer.
 
 #### options
 
-##### progressive
+| Option | Type | Default | Description |
```
Comment (kevva) [inline readme.md]: Please keep the old formatting.

File: index.js [hunk 2]
```diff
@@ -14,14 +14,43 @@ module.exports = opts => buf => {
...
+
 	const args = [
-		'--strip-all',
-		'--strip-iptc',
-		'--strip-icc',
 		'--stdin',
 		'--stdout'
 	];
 
+	// Strip Comment markers from output file
```
Comment (kevva) [inline index.js]: Don't need to comment these ones since they are documented in the readme.

File: index.js [hunk 3]
```diff
@@ -14,14 +14,43 @@ module.exports = opts => buf => {
...
+	opts.stripCom = typeof opts.stripCom === 'undefined' ? true : opts.stripCom;
+	opts.stripExif = typeof opts.stripExif === 'undefined' ? true : opts.stripExif;
+	opts.stripIptc = typeof opts.stripIptc === 'undefined' ? true : opts.stripIptc;
+	opts.stripIcc = typeof opts.stripIcc === 'undefined' ? true : opts.stripIcc;
+	opts.stripXmp = typeof opts.stripXmp === 'undefined' ? true : opts.stripXmp;
+
 	const args = [
-		'--strip-all',
-		'--strip-iptc',
-		'--strip-icc',
```
Comment (kevva) [inline index.js]: Can you add a `stripAll` option too?

File: index.js [hunk 4]
```diff
@@ -14,14 +14,43 @@ module.exports = opts => buf => {
 		return Promise.resolve(buf);
 	}
 
+	// Default is to strip all markers from output file
+	opts.stripCom = typeof opts.stripCom === 'undefined' ? true : opts.stripCom;
+	opts.stripExif = typeof opts.stripExif === 'undefined' ? true : opts.stripExif;
+	opts.stripIptc = typeof opts.stripIptc === 'undefined' ? true : opts.stripIptc;
+	opts.stripIcc = typeof opts.stripIcc === 'undefined' ? true : opts.stripIcc;
+	opts.stripXmp = typeof opts.stripXmp === 'undefined' ? true : opts.stripXmp;
```
Comment (kevva) [inline index.js]: Do this at the top where the `opts` assignment is instead:

```js
opts = Object.assign({
    stripAll: true,
    stripIptc: true,
    stripIcc: true
}, opts);
```

To set individual markers, you'd set `stripAll` to `false` and then enable the ones you want.

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "imagemin-jpegoptim",
-  "version": "5.0.0",
+  "version": "5.1.0",
```
Comment (kevva) [inline package.json]: Don't bump version numbers in PRs. It's up to the author.

Comment (romellem) [inline package.json]: Cool, thanks for all the feedback! I'll work on these changes soon.

File: index.js [hunk 6]
```diff
@@ -15,13 +17,34 @@ module.exports = opts => buf => {
...
-		'--strip-all',
-		'--strip-iptc',
-		'--strip-icc',
 		'--stdin',
 		'--stdout'
 	];
 
+	if (opts.stripAll) {
+		args.push('--strip-all');
+	} else {
```
Comment (kevva) [inline index.js]: We don't need `else` here.

File: readme.md [hunk 7]
```diff
@@ -53,7 +53,51 @@ Set maximum image quality factor. (`0`-`100`).
...
+Default: `true`
+
+Strip all markers from output file. If you want to control what markers are stripped, this _must_ be set to `false`.
+
+##### stripCom
+
+Type: `boolean`  
+Default: `true`
+
+Strip Comment markers from output file. Requires `stripAll` to be `false`.
```
Comment (kevva) [inline readme.md]: Hmm, I think the note under the `stripAll` option is sufficient. Remove the `Requires stripAll to be false.` from these ones.

**Comment by romellem** [comment]
OK, I think we're good to go.

I also made sure to set the new `strip` flags to all be `true` by default (matches documentation).

So calling

    imageminJpegoptim({
        stripAll: false,
        stripIcc: false
    })

Is the same as calling

    imageminJpegoptim({
        stripAll: false,
        stripIcc: false,
        stripCom: true,
        stripExif: true,
        stripIptc: true,
        stripXmp: true
    })

**Comment by kevva** [comment]
Great! Thank you very much ⭐️ 🍻

**Pull request merged**